python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -971,18 +964,18 @@ def _offset_to_nanoseconds_lower_bound(offset: DateOffset) -> int:
     This utility is used to compute the upper bound of the count of timestamps
     given a range of datetime and an offset.
     """
-    nanoseconds_per_day = 24 * 60 * 60 * 1e9
+    nanoseconds_per_day = 24 * 60 * 60 * 10**9
     kwds = offset.kwds
     return (
         kwds.get("years", 0) * (365 * nanoseconds_per_day)
         + kwds.get("months", 0) * (28 * nanoseconds_per_day)
         + kwds.get("weeks", 0) * (7 * nanoseconds_per_day)
         + kwds.get("days", 0) * nanoseconds_per_day
-        + kwds.get("hours", 0) * 3600 * 1e9
-        + kwds.get("minutes", 0) * 60 * 1e9
-        + kwds.get("seconds", 0) * 1e9
-        + kwds.get("milliseconds", 0) * 1e6
-        + kwds.get("microseconds", 0) * 1e3
+        + kwds.get("hours", 0) * 3600 * 10**9
+        + kwds.get("minutes", 0) * 60 * 10**9
+        + kwds.get("seconds", 0) * 10**9
+        + kwds.get("milliseconds", 0) * 10**6
+        + kwds.get("microseconds", 0) * 10**3
```

Comment:
I thought these would fix #12133, but they did not, so I should spin them out I think.